core/integrations/integrations.js

```diff
@@ -0,0 +1,139 @@
+                console.critical(e);
+            }
+        }
+    }
+    cachedIntegrations = list;
+	return list;
+};
+
+exports.setIntegrationConfigs = function(platform) {
+    for (var i = 0; i < selectedIntegrations.length; i++) {
+        selectedIntegrations[i].instance.platform = platform;
+        selectedIntegrations[i].instance.config = platform.config.loadOutputConfig(selectedIntegrations[i].name);
+        if (!selectedIntegrations[i].instance.config.commandPrefix) {
+            selectedIntegrations[i].instance.config.commandPrefix = platform.defaultPrefix;
+        }
+    }
+}
+
+exports.setIntegrations = function (integrations) {
+    if (selectedIntegrations) {
+        throw 'Cannot change integrations when they are already set.';
+    }
+
+    var i = 0;
+    try {
+        for (i = 0; i < integrations.length; i++) {
+            if (!global.coffeescriptLoaded && integrations[i].start.endsWith('.coffee')) {
+                require("coffee-script").register();
+                global.coffeescriptLoaded = true;
+            }
+            integrations[i].instance = require.once(integrations[i].start);
+
+            if (integrations[i].instance.use) {
+                integrations[i].instance = require.once('./hubot/robot.js').use(integrations[i].instance);
+            }
+        }
+        selectedIntegrations = integrations;
+        return true;
+    }
+    catch (e) {
+        console.critical(e);
+        console.error('Loading the output integration file \'' + integrations[i] + '\' failed.' +
+            '\n\nIf this is your file please ensure that it is syntactically correct.');
+        return false;
+    }
+};
+
+exports.startIntegrations = function (callback) {
+    if (!selectedIntegrations || selectedIntegrations.length === 0) {
+        throw 'Integrations must be set before starting';
+    }
+
+    if (started) {
+        throw 'Cannot start when already started.';
+    }
+
+    for (var i = 0; i < selectedIntegrations.length; i++) {
+        try {
+            console.write("Loading integration '" + selectedIntegrations[i].name + "'...\t");
+            selectedIntegrations[i].instance.start(callback);
+            console.info("[DONE]");
+        }
+        catch (e) {
+            console.error("[FAIL]");
+            console.debug("Failed to start output integration '" + selectedIntegrations[i].name + "'.");
+            console.critical(e);
+        }
+    }
+
+    started = true;
+};
+
+exports.stopIntegrations = function() {
+    if (!started) {
+        throw 'Cannot stop integrations if they haven\'t been started.';
+    }
+
+    for (var i = 0; i < selectedIntegrations.length; i++) {
+        try {
+            selectedIntegrations[i].instance.stop();
+        }
+        catch (e) {
+            console.debug("Failed to correctly stop output integration '" + selectedIntegrations[i] + "'.");
+            console.critical(e);
+        }
+    }
+
+    started = false;
+};
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Newline required at end of file but not found.](https://www.codacy.com/app/theprojectshop/Kassy/file/2530928298/issues/source?bid=3282987&fileBranchId=3283570#l139)